### PR TITLE
Docs/ard upstream node

### PR DESCRIPTION
**Pull request by oskarth**
ARD for upstream/LES/ULC stuff, mostly capturing consensus.

In status-react repo as it already has ARD structure and it impacts end users in terms of options etc.

http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions

<blockquote><div><strong><a href="http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions">Blog | Documenting Architecture Decisions | Relevance</a></strong></div><div>Context Architecture for agile projects has to be described and defined differently. Not all decisions will be made at once, nor will all of them be done whe...</div></blockquote>

### CODE REVIEW

**Review by t1MZXxBn6pAAtTqKTy5UotdeMqBw2CfGrUF — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/M27c2X for branch PR-4243

<blockquote><div><strong><a href="https://i.diawi.com/M27c2X">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/fP3UAz for branch PR-4243

<blockquote><div><strong><a href="https://i.diawi.com/fP3UAz">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: doc/decisions/0003-geth-node.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+our efforts on improving the performance on Whisper and in client-side code.
+
+This also means we don't have a full picture of to what extent LES is a
+performance bottleneck for the Status app. It is possible proper use of CHT,
+slowing sync down, vary load depending on if you are on WiFi or not, etc, can
+solve these problems. However, the consensus right now is that ULC is the more
+promising route to take.
+
+However, since ULC isn't ready yet, this means we are relying on Infura. This is
+a bad idea from a bad idea from a decentralization and security point of view.
```
Comment (adambabik) [inline doc/decisions/0003-geth-node.md]: Typo.

File: doc/decisions/0003-geth-node.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+1. Rely on upstream nodes for beta. For the closed beta scheduled to be releases
+Q2 we will provide two ways of accessing the Ethereum network:
+
+- Upstream RPC with Infura (default)
+- Custom upstream node (https://github.com/status-im/status-react/issues/3994)
+
+From a decentralization point of view we need the latter before we can release
+the beta.
+
+2. ULC MVP. Additionally, as soon as beta is released, assess and start to
```
Comment (adambabik) [inline doc/decisions/0003-geth-node.md:57]: As far as I know, ULC requires trusted peers to operate. How is it different from the upstream node in terms of decentralization and security? 

Decentralization aspect can be explained that it's possible to connect to more than one node in contrast to the upstream node.

Comment (oskarth) [reply doc/decisions/0003-geth-node.md:57]: Good point.

@themue @JekaMas @b00ris @arnetheduck do you want to elaborate on what your point of view is here?

Comment (themue) [reply doc/decisions/0003-geth-node.md:57]: Trust is done by config so far, it has to be done explicitly. So my greatest problem with it is the UX. A distributed directory with a kind of trust rating might help. But that's something for long term. Providing own nodes hardcoded as trusted nodes could act as jump start.

**Comment by oskarth** [comment]
@adambabik @themue Addressed your comments. Please review and approve if you are happy with this

Also added consequence "disable LES for beta"

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/k5yDkt for branch PR-4243

<blockquote><div><strong><a href="https://i.diawi.com/k5yDkt">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/4PeuSu for branch PR-4243

<blockquote><div><strong><a href="https://i.diawi.com/4PeuSu">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by goranjovic — APPROVED** [review]

File: doc/decisions/0003-geth-node.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+## Status
+
+proposed
+
+## Context
+
+On the path to releasing a usable beta we've faced a variety of performance
+issues (https://github.com/status-im/ideas/issues/55,
```
Comment (themue) [inline doc/decisions/0003-geth-node.md]: Just nitpicking, can you create a list for better readability?

File: doc/decisions/0003-geth-node.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+2. ULC MVP. Additionally, as soon as beta is released, assess and start to
+   introduce ULC as an experimental non-default mode in the app.
+   
+By doing this we also support the network, since it is still a very new
+protocol. ULC requires trusted peers to operate, and right now trust is
+established by config (list of trusted peers).
+
+Initially we could hardcode these nodes, with option to add your own. Later on
+we can have some form of distributed directory with a rating system.
```
Comment (themue) [inline doc/decisions/0003-geth-node.md]: Add a note that this also depends from adding ULC to `go-ethereum` and other implementations, not only our patched branch. As long as we're the only ULC supporters we'll have to run the trusted nodes on our own.

**Review by themue — APPROVED** [review]
LGTM. two comments.

**Review by adambabik — APPROVED** [review]

**Comment by asemiankevich** [comment]
@oskarth i believe this could pass `to test` queue and be merged )

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/y2rEvR for branch PR-4243

<blockquote><div><strong><a href="https://i.diawi.com/y2rEvR">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/wfymYV for branch PR-4243

<blockquote><div><strong><a href="https://i.diawi.com/wfymYV">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: doc/decisions/0003-geth-node.md (hunk 3 above)
Comment (oskarth) [reply doc/decisions/0003-geth-node.md]: done

File: doc/decisions/0003-geth-node.md (hunk 4 above)
Comment (oskarth) [reply doc/decisions/0003-geth-node.md]: done

**Pull request merged**